### PR TITLE
perf(parser): eat whitespace after line break

### PR DESCRIPTION
**Pull request by overlookmotel**
Uses the `byte_search!` macro introduced in #2352 to consume whitespace after a line break.

### CODE REVIEW

**Comment by overlookmotel** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2353?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #2350** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2350?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2351** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2351?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #2352** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2352?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #2353** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2353?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/2353?utm_source=stack-comment).

**Comment by Boshen** [comment]
I LOVE all the performance improvements.

One nitpick is the amount of macros we are introducing, it may impact compilation speed and hinder other contributors to navigate the code.

It's troublesome but I think we can expand all the code once we are done ... just like how I did it with the visitors https://github.com/oxc-project/oxc/blob/main/crates/oxc_ast/src/visit.rs ... I wrote it manually 😅

**Comment by overlookmotel** [comment]
It's been fun! By our joint efforts, over the past month the `checker.ts` parser benchmark has gone from 533ms to 350ms.

Maybe you can update the website where it says "2x faster than SWC" to "3x faster"!

It's funny, originally I had it all written out by hand, but it was so repetitive I found it hard to follow, so that's why I brought in the macro.

Have you noticed a difference in compilation times? I haven't noticed one, but OXC compilation falls into the "fast enough" category for me, especially after working on SWC which was so hideously slow in that department.

**Comment by Boshen** [comment]
> Maybe you can update the website where it says "2x faster than SWC" to "3x faster"!

https://github.com/oxc-project/bench-javascript-parser-written-in-rust will automatically pick up the latest release via renovate bot, and I'll update the numbers from that benchmark.

**Comment by Boshen** [comment]
> Have you noticed a difference in compilation times? 

We'll need to look at `cargo build --timings`

**Comment by overlookmotel** [comment]
Would you be willing to cut a release at this point? I think the speed boost in parser justifies it, and I'm dying to see the updated comparison to SWC.
